src/main/java/org/example/bookapp/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.example.bookapp.repository;
+
+import java.util.List;
+import org.example.bookapp.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
Don't use `@Autowired` if you have only one constructor
Add `@RequiredArgsConstructor`